test/functional/mempool_package_limits.py

```diff
@@ -6,48 +6,30 @@
 
 from decimal import Decimal
 
-from test_framework.address import ADDRESS_BCRT1_P2WSH_OP_TRUE
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.messages import (
     COIN,
-    CTransaction,
-    CTxInWitness,
-    tx_from_hex,
     WITNESS_SCALE_FACTOR,
 )
-from test_framework.script import (
-    CScript,
-    OP_TRUE,
-)
 from test_framework.util import (
     assert_equal,
 )
 from test_framework.wallet import (
+    MiniWallet,
     bulk_transaction,
-    create_child_with_parents,
-    make_chain,
-    DEFAULT_FEE,
 )
 
 class MempoolPackageLimitsTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
-        self.setup_clean_chain = True
 
     def run_test(self):
-        self.log.info("Generate blocks to create UTXOs")
-        node = self.nodes[0]
-        self.privkeys = [node.get_deterministic_priv_key().key]
-        self.address = node.get_deterministic_priv_key().address
-        self.coins = []
-        # The last 100 coinbase transactions are premature
-        for b in self.generatetoaddress(node, 200, self.address)[:100]:
-            coinbase = node.getblock(blockhash=b, verbosity=2)["tx"][0]
-            self.coins.append({
-                "txid": coinbase["txid"],
-                "amount": coinbase["vout"][0]["value"],
-                "scriptPubKey": coinbase["vout"][0]["scriptPubKey"],
-            })
+        self.wallet = MiniWallet(self.nodes[0])
+        # the pre-mined test framework chain contains coinbase outputs to the
+        # MiniWallet's default address in blocks 76-100 (see method
+        # BitcoinTestFramework._initialize_chain())
+        self.wallet.rescan_utxos()
+        self.generate(self.wallet, 100) # more mature UTXOs are needed
```

Comment:
I am pretty sure none of the utxos are mature, so the comment doesn't apply.

so maybe this can be removed?

If not, you may just mine 200 blocks in one go instead of using cached/non-cached?

Reply:
It matures the cached blocks, so I could phrase that in better way. But you are right, I changed it to mine the neccesary blocks on a clean chain.

Reply:
blocks 76-100 with a height of 200 should already be mature, no?

Reply:
Yes but because those 25 blocks were not enough for the tests, I also used the 24 initially immature blocks by generating another 100 blocks. But what you suggested is more elegant thus I changed it to that.